content/en/docs/making-requests.md

```diff
@@ -205,136 +254,173 @@ if err := decoder.Decode(&myStruct); err != nil {
+
+// ..
+
+resp, err := client.GetObject(context.Background(), &s3.GetObjectInput{
+	// ...
+})
+if err != nil {
+	log.Printf("error: %v", err)
+	return
+}
+
+requestID, ok := middleware.GetRequestIDMetadata(resp.ResultMetadata)
+if !ok {
+	fmt.Println("RequestID not included with request")
+}
+
+fmt.Printf("RequestID: %s\n", requestID)
+```
+
 ## Concurrently Using Service Clients
 
-You can create goroutines that concurrently use the same service client
-to send multiple requests. You can use a service client with as many
-goroutines as you want. However, you cannot concurrently modify the
-service client's configuration and request handlers. If you do, the
-service client operations might encounter race conditions. Define
-service client settings before you concurrently use it.
+You can create goroutines that concurrently use the same service client to send multiple requests. You can use a service
+client with as many goroutines as you want. 
 
-In the following example, an {{% alias=service=S3 %}} service client is used in multiple
-goroutines. The example concurrently outputs all objects in `bucket1`,
-`bucket2`, and `bucket3`, which are all in the same region. To make
-sure all objects from the same bucket are printed together, the example
-uses a channel.
+In the following example, an {{% alias=service=S3 %}} service client is used in multiple goroutines. This example
+concurrently uploads two objects to an {{% alias service=S3 %}} bucket.
 
 ```go
-sess, err := session.NewSession()
+import "log"
+import "github.com/aws/aws-sdk-go-v2/config"
+import "github.com/aws/aws-sdk-go-v2/service/s3"
+
+// ...
+
+cfg, err := config.LoadDefaultConfig()
 if err != nil {
-    fmt.Println("Error creating session ", err)
+	log.Printf("error: %v", err)
+	return
 }
-var wg sync.WaitGroup
-keysCh := make(chan string, 10)
-
-svc := s3.New(sess)
-buckets := []string{"bucket1", "bucket2", "bucket3"}
-for _, bucket := range buckets {
-    params := &s3.ListObjectsInput{
-        Bucket:  aws.String(bucket),
-        MaxKeys: aws.Int64(100),
-    }
-    wg.Add(1)
-    go func(param *s3.ListObjectsInput) {
-        defer wg.Done()
-
-        err = svc.ListObjectsPages(params,
-            func(page *s3.ListObjectsOutput, last bool) bool {
-                // Add the objects to the channel for each page
-                for _, object := range page.Contents {
-                    keysCh <- fmt.Sprintf("%s:%s", *params.Bucket, *object.Key)
-                }
-                return true
-            },
-        )
-        if err != nil {
-            fmt.Println("Error listing", *params.Bucket, "objects:", err)
-        }
-    }(params)
+
+client := s3.NewFromConfig(cfg)
+
+type result struct {
+    Output *s3.PutObjectOutput
+    Err    error
 }
+
+results := make(chan result, 2)
+
+var wg sync.WaitGroup
+wg.Add(2)
+
 go func() {
-    wg.Wait()
-    close(keysCh)
+defer wg.Done()
+    output, err := client.PutObject(context.Background(), &s3.PutObjectInput{
```

Comment:
As a general what do you think abvout using `context.TODO()` here to signal replace user user's context.